### ISSUE TITLE
Add a rendering pipeline (Exporting the layer stack)

### ISSUE THREAD

**Issue by EmilDohne** [EmilDohne/PhotoshopAPI#94]
This feature would add a rendering pipeline that allows for the extraction of the merged image result even after we modify the layer structure. Meaning we add support for 

**A)** Rendering of any layer via some Layer<T>::render() function which extracts the image data for the given layer. For image layers this would be trivial but e.g. text layers would require some sort of text rendering library or for smart objects an image io library.

**B)** Composing of the layers based on the blend modes into a unified image buffer

As for the timing of this issue as its a rather large task I expect this to come out after I implemented support for at least basic adjustment layers and for this to be a continuous effort rather than being fully implemented in a single release. I will likely also split off into multiple subissues once I get started on this.

If there is any considerations, requests etc. please feel free to drop them below :)

**Comment by 593zhangyu** [comment]
Can I hire technical personnel for development, or do you charge commercial fees on your end?

**Comment by EmilDohne** [comment]
Hey @593zhangyu, do you mean hiring me to implement this feature? Or do you mean hiring someone else to take a look at this? Since this is an open source library people are more than welcome to contribute!

**Comment by 593zhangyu** [comment]
> Hey [@593zhangyu](https://github.com/593zhangyu), do you mean hiring me to implement this feature? Or do you mean hiring someone else to take a look at this? Since this is an open source library people are more than welcome to contribute!

Yes, I need this function, but I don't know how much it will cost.

**Comment by EmilDohne** [comment]
I do have some time today to look into this. If I had to guess though I think a feature like this will take me about a week of uninterrupted work. I'll let you know in a bit how it is going!

**Comment by 593zhangyu** [comment]
> That would however only tackle what is already supported. For example filters on layers such as stroke etc. would not be supported by this yet
Sure, looking forward to your quote

**Comment by wellitecho** [comment]
I'm using psd-tools to convert the psd to jpg. 
However, psd.composite(ignore_preview=True, force=True) takes some 15 seconds to render a simple image. It is really slow. If I simply do psd.composite(), it takes around 0.05 seconds. But the output image is all black(which is explained by [issue 179](https://github.com/EmilDohne/PhotoshopAPI/issues/179)

Is there any other faster way to achieve this?

thanks.
